examples/train_softmax.py

```diff
@@ -0,0 +1,149 @@
+    parser.add_argument('--valid_dataset', '-v', type=str, default='data/molecules_valid_qm9.json',
+                        help='Validation dataset')
+    parser.add_argument('--model_name', choices=['DecomposableAttentionModel', 'BindingModel'],
+                        default='BindingModel')
+    parser.add_argument('--merge_molecule_channels', '-m', type=int, default=10,
+                        help='Number of channels to use in the hidden layers')
+    parser.add_argument('--merge_prot_channels', '-p', type=int, default=10,
+                        help='Number of channels to use in the hidden layers')
+    parser.add_argument('--hidden_channels', '-c', nargs='*', type=int, default=[10],
+                        help='Number of channels to use in the hidden layers')
+    parser.add_argument('--conv_kernel_sizes', '-k', nargs='*', type=int, default=None,
+                        help='Number of channels to use in the hidden layers')
+    parser.add_argument('--num_gnn_steps', '-n', type=int, default=3,
+                        help='Number of times to pass graph through GNN')
+    parser.add_argument('--learning_rate', '-l', type=float, default=1e-3,
+                        help='Learning Rate')
+    parser.add_argument('--lmarch', '-a', type=str, default='elmo',
+                        help='Language Model Architecture')
+    parser.add_argument('--cuda', dest='cuda', action='store_true',
+                        help='Use CUDA (default)')
+    parser.add_argument('--no-cuda', '--cpu', dest='cuda', action='store_false',
+                        help='Use CPU')
+    parser.set_defaults(cuda=True)
+    args = parser.parse_args()
+    return args
+
+
+def run_model_on_softmax_batch(model, multi_batch, device='cuda'):
+    outputs = [run_model_on_batch(model, b, device) for b in multi_batch]
+    out_features = torch.cat(outputs, dim=1)
+    return out_features
+
+
+def get_softmax_targets(multi_batch, device):
+    targets = [get_targets(b, device) for b in multi_batch]
+    targets = torch.nonzero(torch.stack(targets, dim=1))[:, 1]
+    return targets
+
+
+def main():
+    args = _parse_args()
+
+    if args.cuda:
+        device = 'cuda'
+    else:
+        device = 'cpu'
+
+    if not os.path.exists(args.dir):
+        os.makedirs(args.dir)
+    writer = initialize_logging(args.dir, '/logs/')
+
+    # Save the construction arguments for future reference.
+    with open(args.dir + '/training_args.pkl', 'wb') as f:
+        pickle.dump(args, f, pickle.HIGHEST_PROTOCOL)
+
+    lm, path = pretrained_language_models[args.lmarch]
+
+    train_dataset = ComparisonDrugProteinDataset(args.train_dataset)
+    valid_dataset = ComparisonDrugProteinDataset(args.valid_dataset)
+
+    cfxn = lambda x: collate_fn_triplet(x, prots_are_sequences=True)
+    train_dataloader = DataLoader(train_dataset, args.batch_size, shuffle=True, collate_fn=cfxn)
+    valid_dataloader = DataLoader(valid_dataset, args.batch_size, shuffle=True, collate_fn=cfxn)
```

Comment:
I bet that we will need `drop_last=True` here, since the last batch in the epoch usually messes things up and changes the dimensions. https://pytorch.org/docs/stable/data.html